protocol/messenger_handler.go

```diff
@@ -187,7 +187,8 @@ func (m *Messenger) HandleMembershipUpdate(messageState *ReceivedMessageState, c
 	// Store in chats map as it might be a new one
 	messageState.AllChats.Store(chat.ID, chat)
 
-	if waitingForApproval {
+	autoJoin := wasUserAdded // explicit join has been removed, mimic auto-join for backward compatibility
```

Comment:
This is a bit tricky, because it can be used as a vector for attack, a spammer can send a lot of these requests and force the device to reply to each one, effectively DOSing the device.
I think we can get away with a slightly different behavior, we can auto-join if:

1) the chat is active (in that case the user must have already opened the chat)
2) When the user accepts the notification on the activity center

That way (if I am not mistaking):

1) If the user is invited by someone running a new build, no backward compatibility problems, as they send to everyone, messages will be saved even if you haven't explicitly joined, as per this code :+1: 

2) Old users will only send messages to this user once they join, and they will see that he has joined, only when they open the chat from the activity center, which is (for them), expected behavior

Not sure I am missing something, what do you think?

Reply:
That's a good point!

> the chat is active (in that case the user must have already opened the chat)

according to this condition: `isActive := messageState.CurrentMessageState.Contact.Added || messageState.CurrentMessageState.Contact.ID == ourKey || waitingForApproval`chat is active if we were added by added contact.  As a consequence we can be spammed by added contacts, but that's acceptable, isn't it? We decide who we add to the contacts and we can remove malicious contacts at any time :thinking: 

Could we simplify the flow and only auto-join only if we are added by a contact? I know it will not cover all scenarios, e.g. when legacy user invites us before we accept contact request, but since it's kind of an edge case, it's not worth complicating the flow.

Reply:
> As a consequence we can be spammed by added contacts, but that's acceptable, isn't it? 
Yes, that's fine I think

> Could we simplify the flow and only auto-join only if we are added by a contact? 
Sure, we can start with that, maybe leave a comment explaining that not all the cases are covered (i.e invited by a non-contact), and we can either address separately or leave it as it is, we can make a decision closer to mobile release (we'd be looking into it, you don't have to), thanks!